src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockMethodsServer.scala

```diff
@@ -16,13 +16,165 @@ import DefaultJsonProtocol._
 
 object MockMethodsServer {
 
+
+
   val methodsServerPort = 8989
 
   val methodsUrl = FireCloudConfig.Agora.authPrefix + FireCloudConfig.Agora.methodsPath
   val configsUrl = FireCloudConfig.Agora.authPrefix + FireCloudConfig.Agora.configurationsPath
 
+
+
+  val allPermsConfPath1="/namespace/mockAgoraOwnerAllFivePermission/1/permissions"
+  val allPermsConfPath2="/namespace/mockAgoraOwnerAllPermission/1/permissions"
+  val readerPermsConfPath="/namespace/mockAgoraReader/1/permissions"
+  val noPermsNothingConfPath="/namespace/mockAgoraNoPermsNothing/1/permissions"
+  val noPermsEmptyConfPath="/namespace/mockAgoraNoPermsEmpty/1/permissions"
+
+
   /****** Mock Data ******/
 
+  def generateRandomFireCloudRole():String = {
+    val randomNum=scala.util.Random.nextInt(2)
+    if(randomNum==0)
+      {
+        return "NO ACCESS"
+      }
+    else if(randomNum==1)
+      {
+        return "READER"
+      }
+    else
+      {
+        return "OWNER"
+      }
+
+  }
+
+
+  def generateRandomFireCloudPermissions():List[FireCloudPermission] ={
+    List.tabulate(randomPositiveInt()-1)(
+      n=>
+        FireCloudPermission(
+          user=Some(randomAlpha()),
+          role=Some(generateRandomFireCloudRole())
+        )
+    )
+  }
+
+
+
+
+
+  def generateRandomAgoraRoles():List[String] =
+  {
+    //each of the 5 individual roles
+    //has a 1/2 probability of showing up
+    val randomNum=scala.util.Random.nextInt(2)
+    //println("A randomNum is "+randomNum)
+    var aList=List[String]()
+    if(randomNum ==0) {
+      aList=(aList :+ "Read")
+    }
+    val randomNum2=scala.util.Random.nextInt(2)
+    if(randomNum2 ==0 )
+    {
+      aList=(aList :+ "Write")
+    }
+    val randomNum3=scala.util.Random.nextInt(2)
+    if(randomNum3  ==0)
+    {
+      aList=(aList :+ "Create")
+    }
+    val randomNum4=scala.util.Random.nextInt(2)
+    if(randomNum4 ==0)
+    {
+      aList=(aList :+ "Redact")
+    }
+    val randomNum5=scala.util.Random.nextInt(2)
+    if(randomNum5 ==0)
+    {
+      aList=(aList :+ "Manage")
+    }
+    return aList
+  }
```

Comment:
This needs to be re-thought. Do we really need this level of complexity for a random list of strings?